iota-client/src/node/mqtt.rs

```diff
@@ -83,9 +96,19 @@ async fn get_mqtt_client(client: &mut Client) -> Result<&mut MqttClient> {
             };
             for node in nodes.iter() {
                 let host = node.host_str().expect("Can't get host from URL");
-                let mut mqttoptions = MqttOptions::new(host, host, 1883);
-                mqttoptions.set_connection_timeout(client.broker_options.timeout.as_secs());
-                let (_, mut connection) = MqttClient::new(mqttoptions.clone(), 10);
+                let id = "iota.rs";
+                let mut port = client.broker_options.port;
+                let mut uri = format!("wss://{}:{}/mqtt", host, node.port_or_known_default().unwrap_or(port));
+                if !client.broker_options.use_ws {
+                    uri = host.to_string();
+                    port = 1883;
```

Comment:
I think `port = 1883;` should be removed now, because it's the default port above and if the user provides another one we should also use the one from the user provided